packages/upload-api/src/access/authorize.js

```diff
@@ -15,6 +17,23 @@ export const provide = (ctx) =>
  * @param {API.AccessServiceContext} ctx
  */
 export const authorize = async ({ capability }, ctx) => {
+  const isBlocked = await areAnyBlocked(
+    ctx.rateLimitsStorage,
+    emailAndDomainFromMailtoDid(
+      /** @type {import('@web3-storage/did-mailto/dist/src/types').DidMailto} */ (
+        capability.nb.iss
+      )
+    )
+  )
+  if (isBlocked.error || isBlocked.ok) {
```

Comment:
If there is an error we seem to disregard it and return `AccountBlocked` error. This seems unexpected and some code comment would be helpful here.

Reply:
yea I went back and forth on this one - I figured it's moderately better to not authorize a new agent if we get an unexpected error - they can always retry, but we can't easily invalidate a set of delegations once we sign off on them. I'll add a code comment for sure, and am definitely open to reconsidering this!